libs/providers/langchain-google-genai/src/utils/common.ts

```diff
@@ -527,7 +527,7 @@ export function mapGenerateContentResultToChatResult(
   }
   const [candidate] = response.candidates;
   const { content: candidateContent, ...generationInfo } = candidate;
-  const functionCalls = candidateContent.parts.reduce((acc, p) => {
+  const functionCalls = candidateContent.parts?.reduce((acc, p) => {
```

Comment:
@hntrl @encodedz 
```suggestion
  const functionCalls = candidateContent?.parts?.reduce((acc, p) => {
```

This is missing one more optional chain as the bug now evolved to:

```
Cannot read properties of undefined (reading 'parts')
```

<img width="901" height="680" alt="Image" src="https://github.com/user-attachments/assets/01fbea37-7d71-42a9-8bed-5d3dae5d8960" />